app/helpers/application_html_formatters_helper.rb

```diff
@@ -74,7 +75,9 @@ def format_code_block(code, language = nil)
       end
     end
 
-    format_with_pipeline(DefaultCodePipeline, code)
+    formatted_code = DefaultCodePipeline.to_document(code)
```

Comment:
This I believe is breaking abstraction. You should be passing this as a pipeline parameter, I believe.

Which means maybe we should not have a constant global pipeline, instead we should be instantiating them as required.